### PR TITLE
Task2 - Stack with generics

### PR DESCRIPTION
**Pull request by silvaby**
**Task:**
Create a new class Stack.
It should have the following interface:

func push(element: Int) // Adds element on the top of the stack.
`func pop() -> Int?` // Removes and returns the top element of the stack. returns nil when the invoking stack is empty.
`func top() -> Int?` // Returns the top element of the stack, but does not remove it.
Create a new class StackStatistics that is a simple stack as described above but it also should support the following operation:
`func minimumElement() -> Int?` // Returns the first minimum element in the stack. Important: the function should work in O(1).

Use inheritance. Use generics.

### CODE REVIEW

File: Task2.playground/Contents.swift [hunk 1]
```diff
@@ -10,71 +10,63 @@
...
-        if minimumElements.last != nil {
-            minimumElements.removeLast()
-        }
+    override func pop() -> Element? {
+        minimumElements.popLast()
         return super.pop()
     }
 
-    func minimumElement() -> Int? {
+    /// Returns the first minimum element in the stack.
```
Comment (AndreyKovalevskiy) [inline Task2.playground/Contents.swift]:
```suggestion
    /// Returns the minimum element in the stack.
```

Comment (AndreyKovalevskiy) [reply Task2.playground/Contents.swift]: I think word `first` can be misleading.

Comment (silvaby) [reply Task2.playground/Contents.swift]: Thanks. This is a description from the task.

Comment (AndreyKovalevskiy) [reply Task2.playground/Contents.swift:58]: @silvaby but it doesn’t mean that you must take this description.
Better make the description clear.

**Comment by silvaby** [comment]
Fixed.

**Pull request merged**